kuma/resource_health_check.go

```diff
@@ -0,0 +1,544 @@
+package kuma
+
+import (
+	"context"
+
+	"github.com/golang/protobuf/ptypes/wrappers"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/kumahq/kuma/pkg/core/resources/apis/mesh"
+
+	mesh_proto "github.com/kumahq/kuma/api/mesh/v1alpha1"
+	core_store "github.com/kumahq/kuma/pkg/core/resources/store"
+)
+
+func resourceHealthCheck() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceHealthCheckCreate,
+		ReadContext:   resourceHealthCheckRead,
+		UpdateContext: resourceHealthCheckUpdate,
+		Schema: map[string]*schema.Schema{
+			"mesh": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"sources": {
+				Type:     schema.TypeSet,
+				Required: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"match": {
+							Type:     schema.TypeMap,
+							Required: true,
+							Elem: &schema.Schema{
+								Type:     schema.TypeString,
+								Optional: true,
+							},
+						},
+					},
+				},
+			},
+			"destinations": {
+				Type:     schema.TypeSet,
+				Required: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"match": {
+							Type:     schema.TypeMap,
+							Required: true,
+							Elem: &schema.Schema{
+								Type:     schema.TypeString,
+								Optional: true,
+							},
+						},
+					},
+				},
+			},
+			"conf": {
+				Type:     schema.TypeMap,
+				Required: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"interval": {
+							Type: schema.TypeString,
+						},
+						"timeout": {
+							Type: schema.TypeString,
+						},
+						"unhealthy_threshold": {
+							Type: schema.TypeString,
+						},
+						"healthy_threshold": {
+							Type: schema.TypeString,
+						},
```

Comment:
It seems there are now more parameters for the health check:
https://kuma.io/docs/1.1.1/policies/health-check/#usage
They are optional tho, so I'd be fine just starting with these, but just for awareness